src/Methods/Modular_Programming/Hard/Q1/RightPrimeTest.java

```diff
@@ -0,0 +1,45 @@
+package Methods.Modular_Programming.Hard.Q1;
+import global.BaseTest;
+import global.variables.*;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.RepetitionInfo;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RightPrimeTest extends BaseTest{
+
+    public Clause[] testSentence() {
+        provideInput("5");
+        return new Clause[]{
+                new StringLiteral("Enter a number"),
+                new NewLine(),
+                new StringLiteral("true"),
+        };
+    }
+
+    public void runMain() {
+        RightPrime.main(new String[]{});
+    }
+
+    @Test
+    public void notPrimeAfterTruncationsTest(){
+        assertFalse(RightPrime.isRightPrime(1277), "Your program does not check the prime after multiple truncations");
+    }
+
+    @Test
+    public void largePrimeTest() {
+        assertTrue(RightPrime.isRightPrime(59393), "Your program does not handle large right truncatable primes");
+    }
+
+    int[] in = {3, 67, 345};
```

Comment:
I feel like https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests could help here

Reply:
Also let's name this variable better, `in` is a bit confusing